clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy

```diff
@@ -226,11 +326,83 @@ class KubernetesClientApiAdapter {
+
+      V1beta1DaemonSet daemonSet = extApi.replaceNamespacedDaemonSet(name, namespace, deployedControllerSet, null)
+      int terminationGracePeriodSeconds = daemonSet.spec.template.spec.terminationGracePeriodSeconds ?: TERMINATION_GRACE_PERIOD_SECONDS
+
+      waitForNumberOfPodUpdated(namespace, daemonSet.metadata.labels, 1, SHUTDOWN_ALL_PODS, terminationGracePeriodSeconds)
+
+      return daemonSet
+    }
+  }*/
+
+  V1beta1DaemonSet replaceDaemonSet(String name, String namespace, V1beta1DaemonSet daemonSet) {
+    exceptionWrapper("DaemonSet.replace", "Replace Daemon Set ${name}", namespace) {
+      def deployedControllerSet = getDaemonSet(name, namespace)
+      return extApi.replaceNamespacedDaemonSet(name, namespace, KubernetesClientApiConverter.toReplaceDaemonSet(daemonSet, deployedControllerSet), API_CALL_RESULT_FORMAT)
+    }
+  }
+
+  V1beta1DaemonSet getDaemonSet(String name, String namespace) {
+    exceptionWrapper("DaemonSet.get", "Get Daemon Set ${name}", namespace) {
+      try {
+        return extApi.readNamespacedDaemonSet(name, namespace, API_CALL_RESULT_FORMAT, true, false)
+      } catch(Exception e) {
+        LOG.info(e.message)
+      }
+      return null
+    }
+  }
+
   List<String> getNamespacesByName() {
     exceptionWrapper("namespaces.list", "Get Namespaces", null) {
       V1NamespaceList result = coreApi.listNamespace(API_CALL_RESULT_FORMAT, null, null, null, null, null, null, 30, null)
       return result.items.collect { n -> n.getMetadata().getName() }
     }
   }
+
+  /*boolean hardDestroyDaemonSet(String name, String namespace) {
+    exceptionWrapper("daemonSets.delete", "Hard Destroy Daemon Set ${name}", namespace) {
+      V1DeleteOptions body = new V1DeleteOptions()
+      Boolean orphanDependents = true
+      def deployedControllerSet = getDaemonSet(name, namespace)
+      V1beta1DaemonSet daemonset = deleteDaemonSetPod(name, namespace, deployedControllerSet)
+      //V1Status status = coreApi.deleteNamespacedPod(name, namespace, body, null, gracePeriodSeconds, orphanDependents, null);
+      V1Status status
+      try {
+        status = extApi.deleteNamespacedDaemonSet(name, namespace, body, null, TERMINATION_GRACE_PERIOD_SECONDS, orphanDependents, null);
+      } catch(Exception e) {
+        LOG.debug(e.message)
+      }
+        return (status?.status == "Success" ? true : false )
+    }
+  }*/
+
+  /*
+  Reference Kubernetes patch operation spec - https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#patch-operations
+  Example,
+    [
+     { "op": "test", "path": "/a/b/c", "value": "foo" },
+     { "op": "remove", "path": "/a/b/c" },
+     { "op": "add", "path": "/a/b/c", "value": [ "foo", "bar" ] },
+     { "op": "replace", "path": "/a/b/c", "value": 42 },
+     { "op": "move", "from": "/a/b/c", "path": "/a/b/d" },
+     { "op": "copy", "from": "/a/b/d", "path": "/a/b/e" }
+   ]
+  */
+  class KubernetesControllerPatch {
```

Comment:
I think you can rely on the json patch library that's imported into this clouddriver module instead

Reply:
For resize, I will keep this implementation becuase it is only one field change. For update, I will implement your suggestion to use patch library.

Reply:
I think it's neater not to expose these patch semantics. It also couples us to this particular patch strategy. You could just as easily fetch the current stateful set, make an edit to that object, and then use that to patch the statefulset.